modules/core/core-internal/src/main/java/com/enonic/xp/core/internal/Exceptions.java

```diff
@@ -0,0 +1,31 @@
+package com.enonic.xp.core.internal;
+
+import java.util.concurrent.ExecutionException;
+
+public final class Exceptions
+{
+    private Exceptions()
+    {
+    }
+
+    public static void throwIfUnchecked( final Throwable throwable )
+    {
+        if ( throwable instanceof RuntimeException )
+        {
+            throw (RuntimeException) throwable;
+        }
+        if ( throwable instanceof Error )
+        {
+            throw (Error) throwable;
+        }
```

Comment:
What about 
```suggestion
        }
        throw new IllegalStateException(throwable);
```

Reply:
This method only throws unchecked exceptions